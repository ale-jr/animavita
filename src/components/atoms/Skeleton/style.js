import styled, { keyframes } from "styled-components";
import PropTypes from "prop-types";

const shine = keyframes`
100%{
  transform: translateX(100%);
}
`;

export const Skeleton = styled.span.attrs({
    children: <>&zwnj;</>,
    'aria-hidden': 'true'
})`
  background-color: ${({ theme }) => theme.colors.neutral.lightest};
  width: ${({ theme, size, width }) =>
        width || (size ? theme.spacing[size] : "100%")};
  height: ${({ theme, size, height }) =>
        height || (size ? theme.spacing[size] : "unset")};

  border-radius: ${({ circle }) => circle ? '500px' : '8px'};

  display: inline-flex;
  line-height: 1;
  position: relative;
  overflow: hidden;

  &::after {
    content: " ";
    display: block;
    position: absolute;
    left: 0;
    right: 0;
    height: 100%;
    background-repeat: no-repeat;
    background-image: linear-gradient(
      90deg,
      ${({ theme }) => theme.colors.neutral.lightest},
      ${({ theme }) => theme.colors.neutral.light},
      ${({ theme }) => theme.colors.neutral.lightest}
    );
    transform: translateX(-100%);
    animation: ${shine} 1s ease-in-out infinite;
  }
`;

Skeleton.defaultProps = {
    circle: false,
};

Skeleton.propTypes = {
    size: PropTypes.oneOf([
        "xxs",
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
    ]),
    circle: PropTypes.bool,
    width: PropTypes.string,
    height: PropTypes.string,
};



export const RepeatSkeletonBox = styled.div`
    display: flex;
    flex-direction: column;
    gap: ${({ theme }) => theme.spacing.xxs};
`;