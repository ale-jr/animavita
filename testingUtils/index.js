import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../src/themes/default";

function ThemeProviderWrapper({ children }) {
  return (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  );
}

export const renderWithTheme = (ui, options) =>
  render(ui, {
    wrapper: ThemeProviderWrapper,
    ...options
  });
