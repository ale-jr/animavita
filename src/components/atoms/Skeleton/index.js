import { Skeleton as BaseSkeleton, RepeatSkeletonBox } from "./style";
import propTypes from "prop-types";

function Skeleton({ repeat, ...otherProps }) {

    if (!repeat) return <BaseSkeleton {...otherProps} />
    else return <RepeatSkeletonBox>
        {Array.from(Array(repeat)).map((_, key) => (<BaseSkeleton key={key} {...otherProps} />))}
    </RepeatSkeletonBox>
}

Skeleton.propTypes = {
    ...BaseSkeleton.propTypes,
    repeat: propTypes.number
}


export default Skeleton
