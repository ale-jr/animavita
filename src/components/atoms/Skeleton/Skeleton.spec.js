import Skeleton from ".";
import { renderWithTheme } from "../../../../testingUtils";

describe("Skeleton", () => {
    it("Matches snapshot", () => {
        expect(renderWithTheme(<Skeleton />).container).toMatchSnapshot()

        expect(renderWithTheme(<Skeleton circle />).container).toMatchSnapshot()

        expect(renderWithTheme(<Skeleton width="100px" height="100px" />).container).toMatchSnapshot()
    })

    it("Renders repeated skeletons", () => {
        const { container } = renderWithTheme(<Skeleton repeat={10} />)
        expect(container).toMatchSnapshot()
    })
})