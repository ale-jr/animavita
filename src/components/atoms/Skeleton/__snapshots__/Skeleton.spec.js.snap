// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton Matches snapshot 1`] = `
<div>
  <span
    aria-hidden="true"
    class="sc-bdvvtL jsLOtp"
  >
    ‌
  </span>
</div>
`;

exports[`Skeleton Matches snapshot 2`] = `
<div>
  <span
    aria-hidden="true"
    class="sc-bdvvtL ismeOg"
  >
    ‌
  </span>
</div>
`;

exports[`Skeleton Matches snapshot 3`] = `
<div>
  <span
    aria-hidden="true"
    class="sc-bdvvtL gUwRCM"
    height="100px"
    width="100px"
  >
    ‌
  </span>
</div>
`;

exports[`Skeleton Renders repeated skeletons 1`] = `
<div>
  <div
    class="sc-gsDKAQ hPaScV"
  >
    <span
      aria-hidden="true"
      class="sc-bdvvtL jsLOtp"
    >
      ‌
    </span>
    <span
      aria-hidden="true"
      class="sc-bdvvtL jsLOtp"
    >
      ‌
    </span>
    <span
      aria-hidden="true"
      class="sc-bdvvtL jsLOtp"
    >
      ‌
    </span>
    <span
      aria-hidden="true"
      class="sc-bdvvtL jsLOtp"
    >
      ‌
    </span>
    <span
      aria-hidden="true"
      class="sc-bdvvtL jsLOtp"
    >
      ‌
    </span>
    <span
      aria-hidden="true"
      class="sc-bdvvtL jsLOtp"
    >
      ‌
    </span>
    <span
      aria-hidden="true"
      class="sc-bdvvtL jsLOtp"
    >
      ‌
    </span>
    <span
      aria-hidden="true"
      class="sc-bdvvtL jsLOtp"
    >
      ‌
    </span>
    <span
      aria-hidden="true"
      class="sc-bdvvtL jsLOtp"
    >
      ‌
    </span>
    <span
      aria-hidden="true"
      class="sc-bdvvtL jsLOtp"
    >
      ‌
    </span>
  </div>
</div>
`;
